ocaml/tests/common/test_common.ml

```diff
@@ -144,20 +144,34 @@ let make_vm ~__context ?(name_label = "name_label")
     ?(hardware_platform_version = 0L) ?has_vendor_device:_
     ?(has_vendor_device = false) ?(reference_label = "") ?(domain_type = `hvm)
     ?(nVRAM = []) ?(last_booted_record = "") ?(last_boot_CPU_flags = [])
-    ?(power_state = `Halted) () =
-  Xapi_vm.create ~__context ~name_label ~name_description ~user_version
-    ~is_a_template ~affinity ~memory_target ~memory_static_max
-    ~memory_dynamic_max ~memory_dynamic_min ~memory_static_min ~vCPUs_params
-    ~vCPUs_max ~vCPUs_at_startup ~actions_after_shutdown ~actions_after_reboot
-    ~actions_after_crash ~pV_bootloader ~pV_kernel ~pV_ramdisk ~pV_args
-    ~pV_bootloader_args ~pV_legacy_args ~hVM_boot_policy ~hVM_boot_params
-    ~hVM_shadow_multiplier ~platform ~nVRAM ~pCI_bus ~other_config
-    ~xenstore_data ~recommendations ~ha_always_run ~ha_restart_priority ~tags
-    ~blocked_operations ~protection_policy ~is_snapshot_from_vmpp ~appliance
-    ~start_delay ~shutdown_delay ~order ~suspend_SR ~suspend_VDI
-    ~snapshot_schedule ~is_vmss_snapshot ~version ~generation_id
-    ~hardware_platform_version ~has_vendor_device ~reference_label ~domain_type
-    ~last_booted_record ~last_boot_CPU_flags ~power_state
+    ?(power_state = `Halted) ?resident_on () =
+  let vm =
+    Xapi_vm.create ~__context ~name_label ~name_description ~user_version
+      ~is_a_template ~affinity ~memory_target ~memory_static_max
+      ~memory_dynamic_max ~memory_dynamic_min ~memory_static_min ~vCPUs_params
+      ~vCPUs_max ~vCPUs_at_startup ~actions_after_shutdown ~actions_after_reboot
+      ~actions_after_crash ~pV_bootloader ~pV_kernel ~pV_ramdisk ~pV_args
+      ~pV_bootloader_args ~pV_legacy_args ~hVM_boot_policy ~hVM_boot_params
+      ~hVM_shadow_multiplier ~platform ~nVRAM ~pCI_bus ~other_config
+      ~xenstore_data ~recommendations ~ha_always_run ~ha_restart_priority ~tags
+      ~blocked_operations ~protection_policy ~is_snapshot_from_vmpp ~appliance
+      ~start_delay ~shutdown_delay ~order ~suspend_SR ~suspend_VDI
+      ~snapshot_schedule ~is_vmss_snapshot ~version ~generation_id
+      ~hardware_platform_version ~has_vendor_device ~reference_label
+      ~domain_type ~last_booted_record ~last_boot_CPU_flags ~power_state
+  in
+  match power_state with
+  | `Running | `Paused ->
+      Db.VM.set_power_state ~__context ~self:vm ~value:power_state ;
+      Option.iter
+        (fun host -> Db.VM.set_resident_on ~__context ~self:vm ~value:host)
+        resident_on ;
+      let vmm = Db.VM.get_metrics ~__context ~self:vm in
+      Db.VM_metrics.set_current_domain_type ~__context ~self:vmm
+        ~value:domain_type ;
+      vm
```

Comment:
This seems unrelated to the rest of the changes, why is updating the power state of the VM, resident_on and metrics needed here, only for vm that have been told to run or pause?